### PR TITLE
Further tweaks

### PR DESCRIPTION
**Pull request by ljedrz**
Builds on https://github.com/AleoHQ/snarkOS/pull/1880.

Contains further tweaks, cleanups and performance improvements, including:
- removing needless clones
- returning handles to long-running tasks
- introducing deferred deserialization (to not block the async executor)
- removing needless re-serialization

### CODE REVIEW

**Review by raychu86 — APPROVED** [review]

**Review by raychu86 — APPROVED** [review]

File: snarkos/ledger/mod.rs [hunk 1]
```diff
@@ -95,36 +95,11 @@ impl<N: Network> Ledger<N> {
         self.ledger.write().add_to_memory_pool(transaction)
     }
 
-    /// Advances the ledger to the next block.
-    pub async fn advance_to_next_block(self: &Arc<Self>) -> Result<Block<N>> {
```
Comment (raychu86) [inline snarkos/ledger/mod.rs]: Unremove this

Comment (ljedrz) [reply snarkos/ledger/mod.rs]: it's unused, though; mark it as `#[allow(dead_code)]`?

Comment (raychu86) [reply snarkos/ledger/mod.rs]: "add_to_memory_pool" is also technically unused by the client then.

We will need to reimplement these later so it is easier to have them here and refactor as oppose to removing them.

**Comment by ljedrz** [comment]
Rebased and applied comments.

**Pull request merged**